### ISSUE TITLE
RCT_EXPORT_METHOD does not parse all Objc-C method declarations

### ISSUE THREAD

**Issue by hermanccw** [facebook/react-native#3804]
Hi all, 

Ran into a problem when trying to integrate RN into an existing app, running version 0.13.2.

Followed instruction here: https://facebook.github.io/react-native/docs/embedded-app-ios.html#content and had a simple view displaying in existing app, great so far.

Problem started when I wanted to have the JS file called into an objc method.

1.  Create RCTBridgeModule class like this: 
```
#import "ReactNativeCalendarManager.h"
#import "RCTLog.h"
@implementation ReactNativeCalendarManager

RCT_EXPORT_MODULE();
RCT_EXPORT_METHOD(addEvent : (NSString *)name location : (NSString *)location)
{
	RCTLogInfo(@"Pretending to create an event %@ at %@", name, location);
}
@end
```

2.  Here is my index.ios.js 
```
 'use strict';
 
 var ReactNativeCalendarManager = require('react-native').NativeModules.ReactNativeCalendarManager;
 
 var React = require('react-native');
 var {
     Text,
     View
 } = React;
 
 var styles = React.StyleSheet.create({
     container: {
         flex: 1,
         backgroundColor: 'white'
     }   
 });
 
 class SimpleApp extends React.Component {
     render() {
         ReactNativeCalendarManager.addEvent('1', '2');
         var dates = []; 
         for (var i = 0; i < 4; i++) {
             dates[i] = []; 
             for (var j = 0; j < 7; j++) {
                 dates[i][j] = i * 7 + 1 + j;
             }   
         }   
         return (
             <View style={styles.container}>
                 {dates.map(function(week) {
                     return <View style={{flexDirection: 'row'}}>
                         {week.map(function(day) {
                             return <Text style={{flex: 1, textAlign: 'center'}}>{day}</Text>;
                         })} 
                     </View>;
                 })} 
                 <Text>{this.state.events}</Text>
             </View>
         )   
     }   
 }
 
 React.AppRegistry.registerComponent('SimpleApp', () => SimpleApp);
```

3.  Got the following error:
```
2015-10-30 17:59:53.813 [warn][tid:com.facebook.React.JavaScript] 'Warning: Native component for "RCTSnapshot" does not exist'
2015-10-30 17:59:53.862 [info][tid:com.facebook.React.JavaScript] 'Running application "SimpleApp" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF'
2015-10-30 17:59:53.880 [warn][tid:com.facebook.React.JavaScript] 'Error: ReactNativeCalendarManager.addEvent is not a function. (In \'ReactNativeCalendarManager.addEvent(\'1\',\'2\')\', \'ReactNativeCalendarManager.addEvent\' is undefined)\n stack: \n  render                                             index.ios.bundle?…:1431\n  _renderValidatedComponentWithoutOwnerOrContext     index.ios.bundle?…:8995\n  _renderValidatedComponent                          index.ios.bundle?…:9017\n  ReactCompositeComponent__renderValidatedComponent  index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:8514\n  ReactCompositeComponent_mountComponent             index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:6722\n  mountChildren                                      index.ios.bundle?…:23399\n  initializeChildren                                 index.ios.bundle?…:19816\n  mountComponent                                     index.ios.bundle?…:19940\n  mountComponent                                     index.ios.bundle?…:6722\n  mountComponent                                     index.ios.bundle?…:8520\n  ReactCompositeComponent_mountComponent             index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:6722\n  mountChildren                                      index.ios.bundle?…:23399\n  initializeChildren                                 index.ios.bundle?…:19816\n  mountComponent                                     index.ios.bundle?…:19940\n  mountComponent                                     index.ios.bundle?…:6722\n  mountComponent                                     index.ios.bundle?…:8520\n  ReactCompositeComponent_mountComponent             index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:6722\n  mountComponent                                     index.ios.bundle?…:8520\n  ReactCompositeComponent_mountComponent             index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:6722\n  mountComponent                                     index.ios.bundle?…:8520\n  ReactCompositeComponent_mountComponent             index.ios.bundle?…:5884\n  mountComponent                                     index.ios.bundle?…:6722\n  mountComponentIntoNode                             index.ios.bundle?…:7479\n  perform                                            index.ios.bundle?…:7123\n  batchedMountComponentIntoNode                      index.ios.bundle?…:7498\n  batchedUpdates                                     index.ios.bundle?…:17024\n  batchedUpdates                                     index.ios.bundle?…:6426\n  renderComponent                                    index.ios.bundle?…:7582\n  ReactMount__renderNewRootComponent                 index.ios.bundle?…:5884\n  render                                             index.ios.bundle?…:1662\n  renderApplication                                  index.ios.bundle?…:46823\n  run                                                index.ios.bundle?…:46713\n  runApplication                                     index.ios.bundle?…:46741\n  __callFunction                                     index.ios.bundle?…:5629\n  <unknown>                                          index.ios.bundle?…:5551\n  guard                                              index.ios.bundle?…:5506\n  <unknown>                                          index.ios.bundle?…:5551\n  <unknown>                                          index.ios.bundle?…:5548\n  perform                                            index.ios.bundle?…:7123\n  batchedUpdates                                     index.ios.bundle?…:17026\n  batchedUpdates                                     index.ios.bundle?…:6426\n  <unknown>                                          index.ios.bundle?…:5547\n  guard                                              index.ios.bundle?…:5506\n  processBatch                                       index.ios.bundle?…:5546\n URL: http://localhost:8081/index.ios.bundle?platform=ios\n line: 1431\n message: ReactNativeCalendarManager.addEvent is not a function. (In \'ReactNativeCalendarManager.addEvent(\'1\',\'2\')\', \'ReactNativeCalendarManager.addEvent\' is undefined)'

```

Basically saying "addEvent" is not defined, any hints?

**Comment by javache** [comment]
Try following the obj-c style for method naming, although this not working sounds like a bug as well

    RCT_EXPORT_METHOD(addEvent:(NSString *)name location:(NSString *)location)

**Comment by hermanccw** [comment]
Thanks @javache!  That's it!  It seems to be still a bug though since that's valid objc.

**Comment by ide** [comment]
There's a regex that parses method signatures that probably needs to be just a tiny bit more lenient. Thanks for the report!

**Comment by nicklockwood** [comment]
I'm about to do some work on this code (to add generics support), so I'll take a look at fixing that issue.

**Comment by nicklockwood** [comment]
Hmm, I can't really explain it, but

    RCT_EXPORT_METHOD(addEvent : (NSString *)name location : (NSString *)location)

works fine for me with the current codebase. Our method matching regex seems to handle extra whitespace around the ":"s without any issue.

**Comment by nicklockwood** [comment]
Ahah! I see the problem. It's calling the method "addEvent ", with a space at the end.

**Comment by ide** [comment]
Fixed in https://github.com/facebook/react-native/commit/6539b268108f1a6976cdda7b393f4abbbe82188e; thanks @nicklockwood.
